extensions/amp-base-carousel/1.0/base-carousel.js

```diff
@@ -191,12 +193,24 @@ function BaseCarouselWithRef(
     return interaction.current === Interaction.TOUCH;
   }, [controls, outsetArrows]);
 
+  const [rtl, setRtl] = useState(false);
+  useLayoutEffect(() => {
+    if (!containRef.current) {
+      return;
+    }
+    const doc = containRef.current.ownerDocument;
+    if (!doc) {
+      return;
+    }
+    setRtl(isRTL(doc));
```

Comment:
I'm wondering if we should auto-discover this on the Preact level or only on AMP level. Overall I'm tending to do it here. But having some doubts since the explicitly requiring `dir` would reduce the chance of FOUC.

Reply:
Would it cause FOUC for the auto case? I thought that when modifying state in `useLayoutEffect`, the component should not be visible.

Reply:
It'd cause FOUC in case of SSR. I.e. if the component is rendered as LTR on the server and then reshuffled on the client as soon as JS arrives and hydrated.

Reply:
Chatted offline, we will keep this as is for now and follow up by supporting `dir` as a standard property in `PreactBaseElement`.